mobility/planner_scp/scp/src/traj_opt.cpp

```diff
@@ -360,872 +914,1150 @@ void TOP::UpdateDynamics() {
-        linear_con_mat.coeffRef(row_idx, state_dim*(ii+1)+kk) = support_vec(kk);
-      }
-
-      row_idx++;
-      col_idx++;
-    }
-  }
-
-  // Slack variables non-negative constraints
-  col_idx = state_dim*N + control_dim*(N-1) +
-    2*7*(N-1) + 2*4*(N-1)  + (state_dim+1)*(N-1);
-  upper_bound.segment(row_idx, n_obs*(N-1)).setZero();
-  for (size_t ii = 0; ii < n_obs*(N-1); ii++) {
-    linear_con_mat.insert(row_idx, col_idx) = -1;
-    row_idx++;
-    col_idx++;
+void TOP::NormalizeQuaternions() {
+  // re-normalize quaternions
+  for (size_t ii = 0; ii < N; ii++) {
+    decimal_t q_norm = Xprev[ii].segment(6, 4).norm();
+    Xprev[ii].segment(6, 4) /= q_norm;
   }
 }
 
-void TOP::UpdateBoundaryCons() {
-  lower_bound.segment(0, state_dim) = x0;
-  upper_bound.segment(0, state_dim) = x0;
-
-  lower_bound.segment(state_dim, state_dim) = xg;
-  upper_bound.segment(state_dim, state_dim) = xg;
-}
-
-void TOP::UpdateControlCons() {
-  size_t row_idx = 2*state_dim;
-
-  for (size_t ii = 0; ii < N-1; ii++) {
-    // Linear accelereation
-    row_idx += 6;
-    upper_bound(row_idx) = mass*desired_accel_;
-
-    row_idx++;
-
-    // Angular acceleration
-    row_idx += 6;
-    Vec3 alpha_;
-    alpha_.setOnes();
-    alpha_ *= desired_alpha_;
-    Vec3 M_ = J*alpha_;
-    upper_bound(row_idx) = M_.minCoeff();
-
-    row_idx++;
+void TOP::PolishSolution() {
+  if (!solved_) {
+    return;
   }
+  NormalizeQuaternions();
 }
 
-void TOP::UpdateStateCons() {
-  // Update any changes to params
-  x_max << pos_max_(0), pos_max_(1), pos_max_(2),
-    desired_vel_/std::sqrt(3), desired_vel_/std::sqrt(3), desired_vel_/std::sqrt(3),
-    1, 1, 1, 1,
-    desired_omega_/std::sqrt(3), desired_omega_/std::sqrt(3), desired_omega_/std::sqrt(3);
-  x_min = -x_max;
-  x_min(0) = pos_min_(0);
-  x_min(1) = pos_min_(1);
-  x_min(2) = pos_min_(2);
-
-  size_t row_idx = 2*state_dim + 20*(N-1);
-
-  // State LB
-  for (size_t ii = 0; ii < N-1; ii++) {
-    // Position limits
-    for (size_t jj = 0; jj < 3; jj++) {
-      upper_bound(row_idx+jj) = -x_min(jj);
-    }
+// NOTE: Functions below this point are not currently being used but may be good for future modularization.
+
+// void TOP::ComputeSignedDistances() {
+//   size_t n_obs = keep_out_zones_->size();
+
+//   collision_checker::SignedDistanceResult sd_result;
+//   for (size_t ii = 0; ii < N-1; ii++) {
+//     for (size_t jj = 0; jj < n_obs; jj++) {
+//       // signed distance, point on co1, point on co2;
+//       cc.ComputeDistance(jj, sd_result);
+//       Vec3 nhat;
+//       if (sd_result.sd >  0) {
+//         nhat = (sd_result.co1_pt-sd_result.co2_pt);
+//       } else {
+//         nhat = (sd_result.co2_pt-sd_result.co1_pt);
+//       }
+//       nhat.normalize();
+//       obs_ub[n_obs*ii+jj] = sd_result.sd - nhat.dot(Xprev[ii].segment(0, 3)) - obs_clearance;
+//       support_vectors[n_obs*ii+jj] = -nhat;
+//     }
+//   }
+// }
 
-    row_idx += 3;
+// void TOP::SetHessianMatrix() {
```

Comment:
Shall we go ahead and delete all the `Set{}Constraints` and methods from the previous SCP architecture if we won't be switching back to them?

Reply:
I admit this is a personal crutch, but it really helps me after writing a piece of constraints code, to go back and check your earlier implementation. I assume the probability of us both getting it right is greater than the probability of us both getting it wrong independently. Of course, I could do this by just saving your earlier code in a separate file. I'm ok with deleting it. 

In the future though, I think going back to the modular function design is a good idea. For now, it makes testing easier to have it in one function.